docs/reference/API/when.md

```diff
@@ -0,0 +1,13 @@
+# `@when`
+
+`@when(event:str = None, selector:str = None)`
+
+The `@when` decorator attaches the decorated function or Callable as an event handler for selected objects on the page. That is, when the named event is emitted by the selected DOM elements, the decorated Python function will be called.
+
+If the decorated function takes a single (non-self) argument, it will be passed the [Event object](https://developer.mozilla.org/en-US/docs/Web/API/Event) corresponding to the triggered event. If the function takes no (non-self) argument, it will be called with no arguments.
+
+## Parameters
+
+`event` - A string representing the event type to match against. This can be any of the [https://developer.mozilla.org/en-US/docs/Web/Events#event_listing](https://developer.mozilla.org/en-US/docs/Web/Events) that HTML elements may emit, as appropriate to their type.
```

Comment:
Nit: shouldn't this be called `event_type` as it literally points at the `event.type` showed in the link?

This is maybe a me-only confusion but since `event` is usually the reference to the *event* object, I've found this name choice a bit off from Web PoV.

Reply:
you did like *and* ignored this comment at the same time ... it's OK to me, no hard feelings 😁

resolving

Reply:
WHOOPS I blew away that progress in a stash/branch change. My bad! Will adjust.